src/common/credential.rs

```diff
@@ -73,49 +79,9 @@ impl Credential {
 
     pub fn tls_config(&self) -> Result<ClientTlsConfig> {
         if let Some(ref tls_root_ca) = self.tls_root_ca {
-            Ok(ClientTlsConfig::new()
-                .ca_certificate(Certificate::from_pem(fs::read_to_string(tls_root_ca)?)))
+            Ok(ClientTlsConfig::new().ca_certificate(Certificate::from_pem(fs::read_to_string(tls_root_ca)?)))
```

Comment:
We could do that. 
This approach avoids internal mutability, and since it's only called once for each cluster node when the connection is opened, I'm not too worried about the performance here. Caching would make credential quite a bit more complex, and I'm not sure that's worth it.